core/controllers/profile.py

```
@@ -334,65 +334,69 @@ def get(self) -> None:
     def put(self) -> None:
         """Handles PUT requests."""
         assert self.user_id is not None
-        update_type = self.payload.get('update_type')
-        data = self.payload.get('data')
+        updates = self.payload.get('updates')
         bulk_email_signup_message_should_be_shown = False
         user_settings = user_services.get_user_settings(self.user_id)
 
-        if update_type == 'subject_interests':
-            self.__validate_data_type(update_type, list, data)
-            user_settings.subject_interests = data
-        elif update_type == 'preferred_language_codes':
-            self.__validate_data_type(update_type, list, data)
-            user_settings.preferred_language_codes = data
-        elif update_type == 'email_preferences':
-            required_keys = [
-                'can_receive_email_updates', 'can_receive_editor_role_email',
-                'can_receive_feedback_message_email',
-                'can_receive_subscription_email']
-            # Check if all required keys are in the data dictionary.
-            if not all(key in data for key in required_keys):
+        for update in updates:
+            update_type = update.get('update_type')
```

These are no longer optional parameters, right? If so, should you raise an error if they aren't provided?

---

Done.